### ISSUE TITLE
`ifNotFloat` has different behavior than `ifNotInt`

### ISSUE THREAD

**Issue by jaredramirez** [rtfeldman/elm-validate#43]
The function `ifNotFloat` will error if the `subject` *is* a float, whereas `ifNotInt` will error if the subject *is not* an int. I believe the behavior of `ifNotInt` is correct. In the definition of `isNotFloat`, `ifTrue`, should be `ifFalse`.

The current definition:
```elm
ifNotFloat : (subject -> String) -> error -> Validator error subject
ifNotFloat subjectToString error =
    ifTrue (\subject -> isFloat (subjectToString subject)) error
```

The correct definition:
```elm
ifNotFloat : (subject -> String) -> error -> Validator error subject
ifNotFloat subjectToString error =
    ifFalse (\subject -> isFloat (subjectToString subject)) error
```

Note:
The definition of `ifNotFloat` and `ifNotInt` also differ in the second argument as well. `ifNotInt` takes a function with the invalid int, whereas `ifNotFloat` takes just the error. Any changes to `ifNotFloat` should probably also reconcile the definitions to be the same.

**Comment by m-przybylski** [comment]
Right. There is also a merge request fixing the issue: https://github.com/rtfeldman/elm-validate/pull/36.

Shame I have spend last week upgrading my elm app from 0.18 to 0.19.1. This time should be spend to rewrite it to react or other front-end js based tool.
